app/modules/code_provider/github/github_service.py

```
@@ -770,3 +792,205 @@ async def check_public_repo(self, repo_name: str) -> bool:
+                )
+
+            # Generate user access token via GitHub App
+            jwt = auth.create_jwt()
+            url = f"https://api.github.com/app/installations/{installation_id}/access_tokens"
+            headers = {
+                "Accept": "application/vnd.github+json",
+                "Authorization": f"Bearer {jwt}",
+                "X-GitHub-Api-Version": "2022-11-28",
+            }
+
+            response = requests.post(url, headers=headers)
```

Ruff (0.13.3)

Probable use of `requests` call without timeout

(S113)